### ISSUE TITLE
More material instantiation bugs

### ISSUE THREAD

**Issue by joelvbernier** [HEXRD/hexrdgui#821]
1. When making a new material, I noticed that the structure factors do not update unless I change the instrument energy.
Example:
- add a new material
- change the space group to 194 (HCP)
- change a = 2.9508, c = 4.6860
- set structure to the following
![image](https://user-images.githubusercontent.com/1154130/111579617-8f4b5600-8773-11eb-8029-8af05cd62069.png)
- now the structure factors appear incorrectly in the reflection table
![image](https://user-images.githubusercontent.com/1154130/111579931-1698c980-8774-11eb-93e1-b688f058164c.png)
- if I toggle the insturement energy in the form view, they get updated to apparently correct values
![image](https://user-images.githubusercontent.com/1154130/111580053-3b8d3c80-8774-11eb-9603-4b6318aa2185.png)

2. If I try to change the space group to cubic (255) from hexagonal (194), I get an `IndexError`
```
Traceback (most recent call last):
  File "c:\users\berni\documents\github\hexrdgui\hexrd\ui\material_editor_widget.py", line 103, in set_space_group
    self.set_material_space_group(sgid)
  File "c:\users\berni\documents\github\hexrdgui\hexrd\ui\material_editor_widget.py", line 159, in set_material_space_group
    self.material.sgnum = sgid
  File "c:\users\berni\documents\github\hexrd\hexrd\material.py", line 696, in _set_sgnum
    self._newUnitcell()
  File "c:\users\berni\documents\github\hexrd\hexrd\material.py", line 234, in _newUnitcell
    self.planeData.lparms = self.reduced_lattice_parameters
  File "c:\users\berni\documents\github\hexrd\hexrd\crystallography.py", line 761, in set_lparms
    self.__calc()
  File "c:\users\berni\documents\github\hexrd\hexrd\crystallography.py", line 627, in __calc
    latPlaneData, latVecOps, hklDataList = PlaneData.makePlaneData(
  File "c:\users\berni\documents\github\hexrd\hexrd\crystallography.py", line 803, in makePlaneData
    latPlaneData = latticePlanes(hkls, lparms,
  File "c:\users\berni\documents\github\hexrd\hexrd\crystallography.py", line 226, in latticePlanes
    L = latticeVectors(lparms, tag)
  File "c:\users\berni\documents\github\hexrd\hexrd\crystallography.py", line 398, in latticeVectors
    cellparms = np.r_[lparms[0], lparms[0], lparms[1],
IndexError: list index out of range
```

**Comment by saransh13** [comment]
@psavery when I perform these functions in the command line, it seems to give the right results. Can you check the update scripts in the GUI?

**Comment by psavery** [comment]
@saransh13 This is fixed by #822 and hexrd/hexrd#230
